ramalama/common.py

```diff
@@ -533,35 +487,44 @@ def set_gpu_type_env_vars():
     get_accel()
 
 
-def get_gpu_type_env_vars():
-    gpu_vars = (
+GPUEnvVar = Literal[
+    "ASAHI_VISIBLE_DEVICES",
+    "ASCEND_VISIBLE_DEVICES",
+    "CUDA_VISIBLE_DEVICES",
+    "HIP_VISIBLE_DEVICES",
+    "INTEL_VISIBLE_DEVICES",
+    "MUSA_VISIBLE_DEVICES",
+]
+
+
+def get_gpu_type_env_vars() -> dict[GPUEnvVar, str]:
```

Comment:
Can we avoid duplication by doing something like this? I didn't test this:

```
GPU_ENV_VARS = (
    "ASAHI_VISIBLE_DEVICES",
    "ASCEND_VISIBLE_DEVICES",
    "CUDA_VISIBLE_DEVICES",
    "HIP_VISIBLE_DEVICES",
    "INTEL_VISIBLE_DEVICES",
    "MUSA_VISIBLE_DEVICES",
)

GPUEnvVar = Literal[
    *GPU_ENV_VARS
]

def get_gpu_type_env_vars() -> dict[GPUEnvVar, str]:
    return {k: os.environ[k] for k in GPU_ENV_VARS if k in os.environ}
```

Reply:
It's tricky we want to maintain compatibility for RHEL9 for the code that runs outside of the container, and that's python3.9 by default... I do wonder is a change like this worth breaking older python3 compatibility, it doesn't enhance our codebase significantly...